aspnet/dnx/projects.rst

```diff
@@ -1,8 +1,652 @@
+
+There is a field in the lock file, ``locked``, which can be set to true either manually or via ``dnu restore —lock``. Setting this field to ``true`` specifies that ``dnu restore`` will just download the versions specified in the lock file and will not do any dependency graph walking or version selection. You can run ``dnu restore --lock`` to generate a locked lock file. Future restores will not change your installed version, unless you use ``dnu restore --unlock`` to remove the lock. You could lock your lock file and check it in on a release branch to ensure that you always get the exact version you expect, but leave it unlocked ()and ignored by source control on development branch(es).
+
+Specifying an alternative package locations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can add ``nupkg`` (NuGet packages) and ``source`` packages (not on a NuGet feed) to a project. To specify the location of these packages you must include the path to the NuGet package or to a packages folder, such as::
+
+	 dnu packages add newPackage.1.0.0.nupkg c:\packageStore  
+	 
+For additional details, run the help command::
+
+	dnu packages add --help
+
+Using commands
+--------------
+
+A command is a named execution of a .NET entry point with specific arguments. You can define commands in your *project.json* file:
+
+.. code-block:: json
+
+    {
+      "version": "1.0.0",
+      "webroot": "wwwroot",
+      "exclude": [
+        "wwwroot"
+      ],
+      "dependencies": {
+        "Kestrel": "1.0.0-beta4",
+        "Microsoft.AspNet.Diagnostics": "1.0.0-beta4",
+        "Microsoft.AspNet.Hosting": "1.0.0-beta4",
+        "Microsoft.AspNet.Server.IIS": "1.0.0-beta4",
+        "Microsoft.AspNet.Server.WebListener": "1.0.0-beta4",
+        "Microsoft.AspNet.StaticFiles": "1.0.0-beta4"
+      },
+      "commands": {
+		"web": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.WebListener --server.urls http://localhost:5000",
+		"kestrel": "Microsoft.AspNet.Hosting --server Kestrel --server.urls http://localhost:5001",
+		"gen": "Microsoft.Framework.CodeGeneration",
+		"ef": "EntityFramework.Commands"
+      },
+      "frameworks": {
+        "dnx451": { },
+        "dnxcore50": { }
+      }
+    }
+
+The commands are a set of arguments that will be passed to the DNX. The first part of this statement must be an assembly with a Main entry point that the DNX will try to execute. 
+	
+You can then use the DNU to install commands globally on a machine::
+
+    dnu commands install MyCommand
+	
+In the ``commands`` section shown above, the EF assembly doesn't require any extra arguments so it is just the name of the assembly.
+
+The assembly listed in the commands should be pulled in by a package that your application depends on.	
```

Comment:
Suggestion: "The entry-point provided by a command should either be the name of a project within your solution, or an assembly provided by a package that your application depends on."